docs/architecture/architecture.md

```diff
@@ -0,0 +1,40 @@
+# OpenFGA High Level Architecture
+
+OpenFGA is a server that reads/writes from a database, and its architecture is similar to any other system that does that:
+
+![basic_architecture](deployment.svg)
+
+- The client application is a service that calls OpenFGA. It can call it using the gRPC API or the HTTP API directly, or through OpenFGA's SDKs (JS, Python, Go, Java, .NET). 
+
+- Calls to the OpenFGA API can be authenticated using a shared secret or through a client credentials flow. If the application uses client credentials, it needs to obtain those for an OAuth client credentials provider (KeyCloak, Auth0, Microsoft Entra etc). 
+
+- The cluster needs to have an ingress for load balancing and eventually service authentication (e.g. nginx).
```

Comment:
> eventually service authentication (e.g. nginx).

Uh? They can do service authentication, but it doesn't need it if using it at the OpenFGA layer.